### PR TITLE
don't violate strict aliasing with reinterpret_cast

### PR DESCRIPTION
**Pull request by tdp2110**
Signed-off-by: Tom Peters <thomas.d.peters@gmail.com>

### CODE REVIEW

**Comment by tdp2110** [comment]
there are some plugins which have the same problem (functions are `writeToBuffer` and `readToBuffer` there). I'll fix those up too

File: plugin/common/plugin.h [hunk 1]
```diff
@@ -57,15 +58,16 @@ class BaseCreator : public IPluginCreator
 template <typename T>
 void write(char*& buffer, const T& val)
 {
-    *reinterpret_cast<T*>(buffer) = val;
+    std::memcpy(buffer, &t, sizeof(T));
```
Comment (rajeevsrao) [inline plugin/common/plugin.h]: &val instead of &t?

File: samples/opensource/sampleUffPluginV2Ext/sampleUffPluginV2Ext.cpp [hunk 2]
```diff
@@ -579,14 +579,15 @@ class UffPoolPluginV2 : public IPluginV2IOExt
     template <typename T>
     void write(char*& buffer, const T& val) const
     {
-        *reinterpret_cast<T*>(buffer) = val;
+        std::memcpy(buffer, &t, sizeof(T));
```
Comment (rajeevsrao) [inline samples/opensource/sampleUffPluginV2Ext/sampleUffPluginV2Ext.cpp]: &val instead of &t?

File: plugin/common/plugin.h (hunk 1 above)
Comment (tdp2110) [reply plugin/common/plugin.h]: absolutely! I'm away from a machine with an nvida graphics card for some time, so I'll have to wait to verify everything (but I'll make this change for sure)

**Review by kevinch-nv — APPROVED** [review]

**Comment by rajeevsrao** [comment]
@tdp2110 can you please incorporate the review changes? Thanks.

**Comment by tdp2110** [comment]
@rajeevsrao I believe it is incorporated.

**Pull request merged**